include/libtorrent/block_cache.hpp

```diff
@@ -337,7 +332,7 @@ namespace libtorrent
 		struct hash_value
 		{
 			std::size_t operator()(cached_piece_entry const& p) const
-			{ return std::size_t(p.storage.get()) + std::size_t(p.piece); }
+			{ return reinterpret_cast<std::size_t>(p.storage.get()) + static_cast<int>(p.piece); }
```

Comment:
yeah, I was wrong. pointer arithmetic is not the right thing here. we want to convert the pointer to an integer and then add to that, not add ``sizeof(storage_interface)``